kolibri/plugins/learn/assets/src/views/content-cards/index.vue

```diff
@@ -0,0 +1,96 @@
+<template>
+
+  <component :header="header" :subheader="subheader" :contents="contentsWithLinks" :is="container" />
+
+</template>
+
+
+<script>
+
+  // this should probably be in the state of parent and passed in as a prop
+  const currentPageMode = require('../../state/getters.js').pageMode;
```

Comment:
I think the easiest thing to do here (and given my comment above, this should probably be something you pass into the card grid and card carousel components as a prop) would be to have a prop that is a `Function` called `genLink` - which then allows the link to be determined by the outer component.